indico/web/forms/fields.py

```diff
@@ -113,6 +116,42 @@ def _value(self):
         return self.data
 
 
+class IndicoLocationField(JSONField):
+
+    CAN_POPULATE = True
+    widget = LocationWidget()
+
+    def __init__(self, *args, **kwargs):
+        self.locations = Location.query.order_by(db.func.lower(Location.name)).all()
+        super(IndicoLocationField, self).__init__(*args, **kwargs)
+
+    def process_formdata(self, valuelist):
+        super(IndicoLocationField, self).process_formdata(valuelist)
+        self.data['room'] = Room.query.get(int(self.data['room_id'])) if self.data.get('room_id') else None
+        self.data['venue'] = Location.query.get(int(self.data['venue_id'])) if self.data.get('venue_id') else None
+
+    def _value(self):
+        if not self.data:
```

Comment:
Here you are checking if the dict is empty and if it is the case then the lines below will cause an error.
```
'address': self.data['address'],
'inheriting': self.data['inheriting']
```